azure-proto-core-test/UserAssignedIdentityTests.cs

```diff
@@ -0,0 +1,377 @@
+            string value = "";
+            using (Stream stream = new MemoryStream())
+            {
+                using (StreamReader streamReader = new StreamReader(stream))
+                {
+                    var writer = new Utf8JsonWriter(stream);
+                    Serialize(writer, userAssignedIdentity);
+                    stream.Seek(0, SeekOrigin.Begin);
+                    value = streamReader.ReadToEnd();
+                }
+            }
+            string actual = "{\"clientId\":\"null\",\"principalId\":\"null\"}";
+            Assert.AreEqual(actual, value);
+        }
+
+        [TestCase]
+        public void TestSerializerNullIdentity()
+        {
+            UserAssignedIdentity userAssignedIdentity = null;
+            using (Stream stream = new MemoryStream())
+            {
+                var writer = new Utf8JsonWriter(stream);
+                Assert.Throws<ArgumentNullException>(delegate { Serialize(writer, userAssignedIdentity); });
+            }
+        }
+
+        [TestCase]
+        public void TestSerializerNullWriter()
+        {
+            UserAssignedIdentity userAssignedIdentity = new UserAssignedIdentity();
+            using (Stream stream = new MemoryStream())
+            {
+                Assert.Throws<ArgumentNullException>(delegate { Serialize(null, userAssignedIdentity); });
+            }
+        }
+
+        [TestCase]
+        public void TestSerializerArray()
+        {
+            UserAssignedIdentity userAssignedIdentity1 = new UserAssignedIdentity(new Guid("3beb288c-c3f9-4300-896f-02fbf175b6be"), new Guid("d0416856-d6cf-466d-8d64-ddc8d7782096"));
+            UserAssignedIdentity userAssignedIdentity2 = new UserAssignedIdentity(new Guid("fbb39377-ff46-4a82-8c47-42d573180482"), new Guid("6d63ce43-c3ac-4b03-933d-4bc31eae50b2"));
+            string value = "";
+            UserAssignedIdentity[] identities = { userAssignedIdentity1, userAssignedIdentity2 };
+            using (Stream stream = new MemoryStream())
+            {
+                using (StreamReader streamReader = new StreamReader(stream))
+                {
+                    foreach (var identity in identities)
+                    {
+                        var writer = new Utf8JsonWriter(stream);
+                        Serialize(writer, identity);
+                        stream.Seek(0, SeekOrigin.Begin);
+                        value = streamReader.ReadToEnd();
+                    }
+                }
+            }
+            string actual = "{\"clientId\":\"3beb288c-c3f9-4300-896f-02fbf175b6be\"," +
+                "\"principalId\":\"d0416856-d6cf-466d-8d64-ddc8d7782096\"}" +
+                "{\"clientId\":\"fbb39377-ff46-4a82-8c47-42d573180482\"," +
+                "\"principalId\":\"6d63ce43-c3ac-4b03-933d-4bc31eae50b2\"}";
+            Assert.AreEqual(actual, value);
+        }
+
+        [TestCase]
+        public void TestEqualsBothIdentitiesNull()
```

Comment:
Not sure how these differ from equality tests above.

Reply:
this is a static Equals function that can be called on two null identities. UserAssignedIdentity.Equals(null,null) is valid, but calling identity1.Equals(identity2) would not work